src/others/pells_equation.md

```diff
@@ -0,0 +1,106 @@
+[//]: # (   $&#40; x + \sqrt{d} \cdot y &#41;$ must be of this form, meaning that all solutions are generated by repeated multiplication of the smallest solution. Here we use Brahmagupta's identity of quadratic decomposition.)
+[//]: # (   $&#40;a^{2} - n \cdot b^{2}&#41; \cdot &#40;c^{2} - n \cdot d^{2}&#41; = &#40;a \cdot c + n \cdot b \cdot d&#41;^{2} - n \cdot &#40;a \cdot d + b \cdot c&#41;^{2}$)
+[//]: # (   )
+[//]: # (Here, $a = x_{1}$, $b = y_{1}$, $c = x_{2}$, $d = y_{2}$)
+[//]: # (   So if &#40;x_{1}, y_{1}&#41; and &#40;x_{2}, y_{2}&#41; are solutions, then $&#40;x_{1} \cdot x_{2} + n \cdot y_{1} \cdot y_{2}, x_{1} \cdot y_{2} + y_{1} \cdot x_{2}&#41;$ is also a solution.)
+[//]: # (First, we prove that product of two solutions is also a solution.)
+
+We use method of descent to prove it. suppose there is a solution $u + v \cdot \sqrt{d}$ such that $u^{2} - d \cdot v^{2} = 1$
+Therefore, $ ( x_{0} + \sqrt{d} \cdot y_{0} )^{n}  < u + v \cdot \sqrt{d} < ( x_{0} + \sqrt{d} \cdot y_{0} )^{n+1}$
+
+Multiplying the above inequality by $( x_{0} - \sqrt{d} \cdot y_{0} )^{n}$,(which is > 0 and < 1) we get
+
+$ 1 < (u + v \cdot \sqrt{d})( x_{0} - \sqrt{d} \cdot y_{0} )^{n} < ( x_{0} + \sqrt{d} \cdot y_{0} ) $
+Because both $(u + v \cdot \sqrt{d})$ and $( x_{0} - \sqrt{d} \cdot y_{0} )^{n}$ have norm 1, their product is also a solution.
+But this contradicts our assumption that $( x_{0} + \sqrt{d} \cdot y_{0} )$ is the smallest solution. Therefore, there is no solution between $( x_{0} + \sqrt{d} \cdot y_{0} )^{n}$ and $( x_{0} + \sqrt{d} \cdot y_{0} )^{n+1}$.
+
+Hence, we conclude that all solutions are given by $( x_{0} + \sqrt{d} \cdot y_{0} )^{n}$ for some integer $n$.
+
+## To find the smallest positive solution
+### Expressing the solution in terms of continued fractions
+We can express the solution in terms of continued fractions. The continued fraction of $\sqrt{d}$ is periodic. Let's assume the continued fraction of $\sqrt{d}$ is $[a_{0}; \overline{a_{1}, a_{2}, \ldots, a_{r}}]$. The smallest positive solution is given by the convergent $[a_{0}; a_{1}, a_{2}, \ldots, a_{r}]$ where $r$ is the period of the continued fraction.
+
+The convergents $p_{n}/q_{n}$ are the rational approximations to $\sqrt{d}$ obtained by truncating the continued fraction expansion at each stage. These convergents can be computed recursively.
+
+Check whether the convergent satisfies the Pell's equation. If it does, then the convergent is the smallest positive solution.
+
+Let's take an example to understand this by solving the equation $x^{2} - 2 \cdot y^{2} = 1$.
+$\sqrt{2} = [1; \overline{2}] = 1 + 1/(2 + 1/(2 + 1/(2+ ...))) $. The convergents are $1/1, 3/2, 7/5, 17/12, 41/29, 99/70, \ldots$.
+Now check for each convergent whether it satisfies the Pell's equation. The smallest positive solution is $3/2$.
+
+### How to calculate the continued fraction of $\sqrt{d}$?
+Let's find the continued fraction of $\def\sf{\sqrt 7}\sf$.
```

Comment:
The definition is broken in github previewer. Please just write \sqrt 7 unless you have a good reason to do a \define.

Reply:
@jxu we don't really use GitHub previewer, as it breaks a lot of stuff (e.g. admonitions) and its rendering is inconsistent with CP-Algorithms anyway. Suggested approach is to use the [preview page](https://cp-algorithms.com/preview.html), or generate a preview on https://gh.cp-algorithms.com/1388/ (done automatically when you click on "Approve and run" for workflows).